mitosheet/src/mito/components/Mito.tsx

```diff
@@ -713,6 +721,8 @@ export const Mito = (props: MitoProps): JSX.Element => {
                     setCurrPathParts={setCurrPathParts}
     
                     failedReplayData={uiState.currOpenTaskpane.failedReplayData}
+
+                    overwriteAnalysisToReplayToMitosheetCall={props.jupyterUtils?.overwriteAnalysisToReplayToMitosheetCall}
```

Comment:
Instead of passing these potentially undefined utilities throughout the app, maybe we should handle the undefined case in one place here. Would make types easier everywhere else.

Reply:
Currently if these are undefiend, I can keep using Mito, which is risky. Someone uses Mito for while and never gets an analysis_id or generated code, so they lose all of their work. 

There used to be a similar problem if the analysis_id had spaces around it, for example, and it caused users to lose their work!

Reply:
There are many cases where you _don't want_ the analysis ID to be written! Namely, if you are not in Juypter, you don't want the frontend to write this ID at all. 

In our streamlit case, the author of the streamlit app is going to be responsible for configuring how the analysis gets saved. But this is to say: it is both valid and legit to have these undefined and not what to do anything with them!